### PR TITLE
Move web structure out of clj web and into cljs nr

### PR DESCRIPTION
**Pull request by jwarwick**
Moved most of the web structure into the reagent/cljs code. We should now be able to modify more of the frontend without needed a server restart.
Made all pages render the same way. All pages (except the game board) are unloaded when you leave them, so should make the front-end snappier.

### CODE REVIEW

File: src/cljs/nr/deckbuilder.cljs [hunk 1]
```diff
@@ -959,21 +959,23 @@
...
-              (end-delete s)
-              (set-deck-on-state s deck))))
-      (fn []
-        [:div.deckbuilder.blue-shade.panel
-         [:div.viewport {:ref #(swap! db-dom assoc :viewport %)}
-          [list-panel s user decks decks-loaded]
-          [selected-panel s]
-          [edit-panel s]]]))))
+
+    (go (while true
```
Comment (NoahTheDuke) [inline src/cljs/nr/deckbuilder.cljs:963]: Won't this happen regardless of which page you're on now?

File: src/cljs/nr/help.cljs [hunk 2]
```diff
@@ -373,7 +373,8 @@
                  content]))))))
 
 (defn help []
+  [:div.page-container
```
Comment (NoahTheDuke) [inline src/cljs/nr/help.cljs:376]: spacing

File: src/cljs/nr/main.cljs [hunk 3]
```diff
@@ -57,27 +64,72 @@
...
+       [news]
+       [chat]
+       [:div#version [:span (str "Version " (get-data "version"))]]])))
+
+(defn pages []
+  (r/create-class
+    {:display-name "main-pages"
+
+     :component-did-mount
+     (fn [] (.setToken history (first (:active-page app-state "/"))))
```
Comment (NoahTheDuke) [inline src/cljs/nr/main.cljs:81]: Will calling `first` on `"\"` work?

**Comment by NoahTheDuke** [comment]
This is glorious, dude. Thank you so much.

**Comment by jwarwick** [comment]
Ha. Actually this is a giant train wreck. But at least it will be a consistent train wreck.

**Pull request merged**